hazelcast/src/test/java/com/hazelcast/jet/core/JobUploadSuccessTest.java

```diff
@@ -67,107 +64,68 @@ public void sha256() throws IOException, NoSuchAlgorithmException {
+                .setJarPath(getJarPath());
 
         jetService.submitJobFromJar(submitJobParameters);
 
         assertEqualsEventually(() -> jetService.getJobs().size(), 1);
-        hazelcastInstance.shutdown();
     }
 
     @Test
-    public void test_jarUploadByClient_withMainClassname() {
+    public void test_jarUploadByNonSmartClient_whenResourceUploadIsEnabled() {
         Config config = smallInstanceConfig();
         JetConfig jetConfig = config.getJetConfig();
         jetConfig.setResourceUploadEnabled(true);
 
-        HazelcastInstance hazelcastInstance = createHazelcastInstance(config);
-        HazelcastInstance client = createHazelcastClient();
+        createHazelcastInstances(config, 2);
+
+        ClientConfig clientConfig = new ClientConfig();
+        ClientNetworkConfig networkConfig = clientConfig.getNetworkConfig();
+        networkConfig.setSmartRouting(false);
+        HazelcastInstance client = createHazelcastClient(clientConfig);
         JetService jetService = client.getJet();
-        List<String> jobParameters = emptyList();
 
-        SubmitJobParameters submitJobParameters = new SubmitJobParameters();
-        submitJobParameters.setJarPath(getJarPath());
-        submitJobParameters.setMainClass("org.example.Main");
-        submitJobParameters.setJobParameters(jobParameters);
+        SubmitJobParameters submitJobParameters = new SubmitJobParameters()
+                .setJarPath(getJarPath());
 
         jetService.submitJobFromJar(submitJobParameters);
 
         assertEqualsEventually(() -> jetService.getJobs().size(), 1);
-        hazelcastInstance.shutdown();
     }
 
     @Test
-    public void test_jarUploadByClient_withWrongMainClassname() {
+    public void test_jarUploadByClient_withMainClassname() {
         Config config = smallInstanceConfig();
         JetConfig jetConfig = config.getJetConfig();
         jetConfig.setResourceUploadEnabled(true);
 
-        HazelcastInstance hazelcastInstance = createHazelcastInstance(config);
+        createHazelcastInstance(config);
         HazelcastInstance client = createHazelcastClient();
         JetService jetService = client.getJet();
         List<String> jobParameters = emptyList();
 
-        SubmitJobParameters submitJobParameters = new SubmitJobParameters();
-        submitJobParameters.setJarPath(getJarPath());
-        submitJobParameters.setMainClass("org.example.Main1");
-        submitJobParameters.setJobParameters(jobParameters);
+        SubmitJobParameters submitJobParameters = new SubmitJobParameters()
+                .setJarPath(getJarPath())
+                .setMainClass("org.example.Main")
+                .setJobParameters(jobParameters);
 
-        assertThrows(JetException.class, () -> jetService.submitJobFromJar(submitJobParameters));
-        hazelcastInstance.shutdown();
+        jetService.submitJobFromJar(submitJobParameters);
 
+        assertEqualsEventually(() -> jetService.getJobs().size(), 1);
```

Comment:
IMHO Too weak assertion. We should check if the job completes (no exceptions, e.g. related to classloading) and that Jar is deleted after successful completion

Reply:
The job is a stream. It does not complete. But I have  added a new assertion to test the job status is running and jar is deleted